pkg/kfilefields/kfilefields.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package kfilefields provides functions to read kernel "struct file" fields against a file descriptor.
+//
+// This is done:
+//   - without using bpf iterators in order to work on old kernels.
+//   - without comparing pids from userspace and ebpf in order to work from
+//     different pid namespaces.
+package kfilefields
+
+import "fmt"
+
+// ReadPrivateDataFromFd uses ebpf to read the private_data pointer from the
+// kernel "struct file" associated with the given fd.
+func ReadPrivateDataFromFd(fd int) (uint64, error) {
+	t, err := creatAndInstallTracer()
+	if err != nil {
+		return 0, fmt.Errorf("creating and installing tracer: %w", err)
+	}
```

Comment:
You mean to avoid an extra call [here](https://github.com/inspektor-gadget/inspektor-gadget/blob/f6569d97981748e999fb476da79937f72613df98/pkg/kfilefields/kfilefields.go#L46) ? Currently it isn't used often so should be fine as it is or am I missing something?

Reply:
We call `ReadPrivateDataFromFd` only once when the `container-hook` package is initialized, so I don't think it makes sense to optimize.

And about `ReadFOpForFdType`: it is used in `specUpdateAddresses()` but there is already a cache, so we won't install the tracer several times.